sklearn/metrics/tests/test_common.py

```diff
@@ -1088,6 +1089,12 @@ def test_sample_weight_invariance(n_samples=50):
                    metric, y_true, y_pred)
 
 
+def test_sample_weight_invariance(n_samples=50):
+    # iterate through each metric testing each case
+    for metrics in generate_sample_weight_invariance(n_samples):
+        pass
```

Comment:
I don't understand the original issue as I commented there and I don't think you need this since nose supports yield in test functions ...